.github/workflows/pmm-ui-tests-upgrade.yml

```diff
@@ -138,111 +145,124 @@ jobs:
+          sudo npx ts-node ./integration-setup.ts --ci --setup-docker-pmm-server --rbac --pmm-server-docker-tag=${{ steps.pmm_server_image.outputs.result }} --pmm-client-version=${{ inputs.pmm_client_start_version }}          
+          timeout 100 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost/ping)" != "200" ]]; do sleep 5; done' || false
 
-      - name: Setup Client for PMM-Server
+      - name: 'Setup <PMM Client: ${{ inputs.pmm_client_start_version }}> and Services'
         run: |
-          sudo bash ./pmm-qa/pmm-tests/pmm2-client-setup.sh --pmm_server_ip 127.0.0.1 --client_version ${{ env.PMM_CLIENT_VERSION }} --admin_password admin --use_metrics_mode no
+          # TODO: un-hardcode pmm server ip and admin password
+          sudo bash ./pmm-qa/pmm-tests/pmm2-client-setup.sh --pmm_server_ip 127.0.0.1 --client_version ${{ inputs.pmm_client_start_version }} --admin_password admin --use_metrics_mode no
           sudo bash ./pmm-qa/pmm-tests/pmm-framework.sh \
             --download \
-            ${{ env.PMM_CLIENTS }} \
+            ${{ inputs.services_list }} \
             --pmm2 \
             --pmm2-server-ip=127.0.0.1
+          # TODO: change sleep into fluent wait
           sleep 30
           sudo pmm-admin list
         shell: bash
 
-      - name: Enable Repository.
-        run: |
-            docker exec pmm-integration-server bash -c "echo exclude=mirror.es.its.nyu.edu | tee -a /etc/yum/pluginconf.d/fastestmirror.conf"
-            docker exec pmm-integration-server yum update -y percona-release
-            docker exec pmm-integration-server sed -i'' -e 's^/release/^/testing/^' /etc/yum.repos.d/pmm2-server.repo
-
-            if [[ ${{ inputs.repository }} == testing ]]; then
-              docker exec pmm-integration-server percona-release enable-only percona testing
-              sudo percona-release enable-only percona testing
-              sudo apt-get update -y
-            elif [[ ${{ inputs.repository }} == experimental ]]; then
-              docker exec pmm-integration-server percona-release enable-only percona experimental
-              sudo percona-release enable-only percona experimental
-              sudo apt-get update -y
-            else
-              docker exec pmm-integration-server percona-release enable-only percona release
-              sudo percona-release enable-only percona release
-              sudo apt-get update -y
-            fi
-
-      - name: Execute Pre Upgrade UI tests
+      - name: 'Run tests before upgrade'
         id: ui-tests-flagged
-        working-directory: pmm-ui-tests
+        working-directory: ./pmm-ui-tests/playwright-tests
         run: |
-          npm install
-          npx playwright install
-          npx playwright test --config="playwright-tests/playwright.config.ts" --grep="${{ inputs.pmm_pre_upgrade_flag }}"
+          touch portalCredentials
```

Comment:
Please remove this, it should not be needed anymore.

Reply:
done.